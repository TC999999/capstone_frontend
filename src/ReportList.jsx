import ReportCard from "./ReportCard.jsx";

const ReportList = ({ reports }) => {
  return (
    <div className="report-list-div">
      {!reports.length ? (
        <p>All reports cleared!</p>
      ) : (
        <ul>
          {reports.map((report) => {
            return (
              <li key={`report-${report.id}`}>
                <ReportCard report={report} />
              </li>
            );
          })}
        </ul>
      )}
    </div>
  );
};

export default ReportList;
