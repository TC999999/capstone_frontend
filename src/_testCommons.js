const TEST_USER = {
  username: "u1",
  firstName: "fn1",
  lastName: "ln1",
  email: "testemail1@gmail.com",
  address: "100 address",
  zipCode: "11111",
  regionOrState: "state1",
  city: "city1",
  country: "country1",
  latitude: "10.99876",
  longitude: "-99.99981",
  isAdmin: false,
  isFlagged: false,
  items: [
    {
      itemID: 1,
      name: "i1",
      initialPrice: 100,
      imageURL: "test_url_1",
      condition: "great",
      description: "test item",
    },
    {
      itemID: 2,
      name: "i2",
      initialPrice: 100,
      imageURL: "test_url_2",
      condition: "great",
      description: "test item",
    },
  ],
  reviews: [],
  reports: [],
  pastPurchases: [
    {
      purchaseID: 1,
      itemName: "i4",
      sellerUser: "u3",
      imageURL: "test_url_4",
      price: 150,
      soldAt: expect.any(Date),
      fromFlaggedUser: false,
    },
  ],
};

const TEST_ITEM = {
  id: 1,
  name: "i1",
  imageURL: "test_url_1",
  initialPrice: 100,
  condition: "great",
  description: "test item",
  sellerUser: "u1",
  isSold: false,
  types: [{ id: 1, name: "electronics" }],
  location: {
    city: "city1",
    address: "100 address",
    latitude: "10.99876",
    longitude: "-99.99981",
    regionOrState: "state1",
    country: "country1",
    zipCode: "11111",
  },
};

const TEST_ITEMS = [
  {
    id: 3,
    name: "i3",
    imageURL: "test_url_3",
    initialPrice: 100,
    condition: "great",
    description: "test item",
    sellerUser: "u2",
    types: [
      { id: 1, name: "electronics" },
      { id: 3, name: "books" },
    ],
    location: {
      city: "city2",
      regionOrState: "state2",
      country: "country2",
      zipCode: "22222",
    },
  },
  {
    id: 2,
    name: "i2",
    imageURL: "test_url_2",
    initialPrice: 100,
    condition: "great",
    description: "test item",
    sellerUser: "u1",
    types: [
      { id: 1, name: "electronics" },
      { id: 2, name: "movies" },
    ],
    location: {
      city: "city1",
      regionOrState: "state1",
      country: "country1",
      zipCode: "11111",
    },
  },
  {
    id: 1,
    name: "i1",
    imageURL: "test_url_1",
    initialPrice: 100,
    condition: "great",
    description: "test item",
    sellerUser: "u1",
    types: [{ id: 1, name: "electronics" }],
    location: {
      city: "city1",
      regionOrState: "state1",
      country: "country1",
      zipCode: "11111",
    },
  },
];

const TEST_MESSAGE = {
  id: 3,
  from: "u2",
  to: "u1",
  itemID: 1,
  itemName: "i3",
  body: "test body",
  createdAt: "2024-07-09 14:23:43.469126-04",
};

const TEST_MESSAGE_2 = {
  id: 3,
  from_username: "u2",
  to_username: "u1",
  itemID: 1,
  item_name: "i3",
  body: "test body",
  sent_at: "2024-07-09 14:23:43.469126-04",
};

const TEST_MESSAGES = {
  conversation: [
    {
      id: 2,
      from_username: "u3",
      to_username: "u1",
      item_name: "i4",
      itemID: 4,
      body: "test body",
      sent_at: "2024-07-09 14:23:43.469126-04",
    },
    {
      id: 1,
      from_username: "u1",
      to_username: "u3",
      item_name: "i4",
      itemID: 4,
      body: "test body",
      sent_at: "2024-07-09 14:23:43.469126-04",
    },
  ],
};

const TEST_REVIEW = {
  id: 1,
  reviewed: "u3",
  reviewer: "u1",
  rating: 10,
  body: "test review",
  madeAt: "2024-07-09 14:23:43.469126-04",
};

const TEST_REPORT = {
  id: 1,
  reportedUser: "u2",
  reporterUser: "u1",
  body: "test body",
  madeAt: "2024-07-09 14:23:43.469126-04",
  isCleared: false,
};

const TEST_REPORTS = [
  {
    id: 1,
    reportedUser: "u2",
    reporterUser: "u1",
    body: "test body",
    madeAt: "2024-07-09 14:23:43.469126-04",
    isCleared: false,
  },
  {
    id: 2,
    reportedUser: "u1",
    reporterUser: "u2",
    body: "test body",
    madeAt: "2024-07-09 14:23:43.469126-04",
    isCleared: false,
  },
];

const TEST_PURCHASE = {
  purchaseID: 1,
  itemName: "i4",
  sellerUser: "u3",
  imageURL: "test_url_4",
  price: 150,
  soldAt: "2024-07-09 14:23:43.469126-04",
  fromFlaggedUser: false,
};

module.exports = {
  TEST_USER,
  TEST_ITEM,
  TEST_ITEMS,
  TEST_MESSAGE,
  TEST_MESSAGE_2,
  TEST_MESSAGES,
  TEST_REPORT,
  TEST_REPORTS,
  TEST_REVIEW,
  TEST_PURCHASE,
};
