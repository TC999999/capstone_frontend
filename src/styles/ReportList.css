.report-list-page-div {
  margin: 5px;
}

.report-list-item {
  list-style-type: none;
  border: 2px solid black;
  padding: 10px;
  margin: 2px;
  background-color: rgb(245, 115, 115);
  box-shadow: 5px 5px 5px black;
}

.report-list-page-div a {
  text-decoration-line: none;
  font-weight: bold;
  font-size: 15px;
  color: blue;
}

.report-list-page-div a:hover {
  text-decoration-line: underline;
  color: purple;
}

.report-list-page-div a:active {
  color: aqua;
}

@media (min-width: 800px) {
  .report-list-page-div a {
    font-size: 20px;
  }
}
