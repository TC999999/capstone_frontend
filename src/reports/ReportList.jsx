import { Link } from "react-router-dom";
import "../styles/ReportList.css";

const ReportList = ({ reports }) => {
  return (
    <div className="report-list-page-div">
      {!reports.length ? (
        <p>All reports cleared!</p>
      ) : (
        <div className="report-list-div">
          {reports.map((report) => {
            return (
              <div className="report-list-item" key={`report-${report.id}`}>
                <Link to={`/reports/${report.id}`}>
                  Report on {report.reportedUser} by {report.reporterUser}
                </Link>
                <p>
                  <small>Made At: {report.madeAt}</small>
                </p>
              </div>
            );
          })}
        </div>
      )}
    </div>
  );
};

export default ReportList;
