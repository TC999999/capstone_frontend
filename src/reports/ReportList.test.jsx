import { render } from "@testing-library/react";
import ReportList from "./ReportList.jsx";
import { TEST_REPORTS } from "../_testCommons.js";
import { MemoryRouter } from "react-router-dom";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <ReportList reports={TEST_REPORTS} />
    </MemoryRouter>
  );
});

it("should match snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <ReportList reports={TEST_REPORTS} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should have report-list", function () {
  const { container } = render(
    <MemoryRouter>
      <ReportList reports={TEST_REPORTS} />
    </MemoryRouter>
  );
  const reports = container.querySelector(".report-list-page-div");
  expect(reports).toBeInTheDocument();
});
