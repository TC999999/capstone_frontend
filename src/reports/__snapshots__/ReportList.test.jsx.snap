// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="report-list-page-div"
  >
    <div
      class="report-list-div"
    >
      <div
        class="report-list-item"
      >
        <a
          href="/reports/1"
        >
          Report on u2 by u1
        </a>
        <p>
          <small>
            Made At: 2024-07-09 14:23:43.469126-04
          </small>
        </p>
      </div>
      <div
        class="report-list-item"
      >
        <a
          href="/reports/2"
        >
          Report on u1 by u2
        </a>
        <p>
          <small>
            Made At: 2024-07-09 14:23:43.469126-04
          </small>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
